src/webots/nodes/WbJoint.cpp

```diff
@@ -292,3 +292,34 @@ void WbJoint::updateJointAxisRepresentation() {
   mMesh = wr_static_mesh_line_set_new(2, vertices, NULL);
   wr_renderable_set_mesh(mRenderable, WR_MESH(mMesh));
 }
+
+const QString WbJoint::urdfName() const {
+  if (positionSensor() && positionSensor()->findSFString("name"))
+    return positionSensor()->findSFString("name")->value();
+  return WbBaseNode::urdfName();
+}
+
+void WbJoint::writeExport(WbVrmlWriter &writer) const {
+  if (writer.isUrdf() && solidEndPoint()) {
+    WbNode *parentRoot = findUrdfLinkRoot();
+    WbVector3 translation = solidEndPoint()->translationFrom(parentRoot);
+    WbVector3 rotationEuler = solidEndPoint()->rotationMatrixFrom(parentRoot).toEulerAngles();
+
+    writer << QString("  <joint name=\"%1\" type=\"continuous\">\n").arg(urdfName());
+    writer << QString("    <parent link=\"%1\"/>\n").arg(parent()->urdfName());
+    writer << QString("    <child link=\"%1\"/>\n").arg(solidEndPoint()->urdfName());
+    writer << QString("    <axis xyz=\"%1 %2 %3\"/>\n").arg((int)axis().x()).arg((int)axis().y()).arg((int)axis().z());
+    writer << QString("    <origin xyz=\"%1 %2 %3\" rpy=\"%4 %5 %6\" />\n")
+                .arg(translation.x())
+                .arg(translation.y())
+                .arg(translation.z())
+                .arg(rotationEuler.x())
+                .arg(rotationEuler.y())
+                .arg(rotationEuler.z());
+    writer << QString("  </joint>\n");
```

Comment:
Instead of manually adding spaces to indent, the `WbVrmlWriter` has these two functions:
```
void increaseIndent()
void decreaseIndent()
```

Reply:
It is a bit messy now, but I guess it is easier to extend and change indentation size

Reply:
Yes indeed, using the increaseIndent/decreaseIndent adds more lines, but is is simpler for the long term maintenance and avoid having to count exactly the number of space for each line.